src/main/java/com/hbm/crafting/handlers/CargoShellCraftingHandler.java

```diff
@@ -56,6 +56,9 @@ public ItemStack getCraftingResult(InventoryCrafting inventory) {
 				copy.stackSize = 1;
 				cargo = copy;
 			}
+			if(stack.getItem() == ModItems.disperser_canister) {
+				stack.getItem().setContainerItem(null); //removes the empty canister from the output
```

Comment:
doing it like this will make it so that the disperser canister won't have a container item at all anymore after this recipe is used once